src/python/WMCore/MicroService/Unified/RequestInfo.py

```diff
@@ -233,41 +230,54 @@ def getParentDatasets(self, workflows):
                 if wflow.hasParents() and wflow.getInputDataset() in parentageMap:
                     wflow.setParentDataset(parentageMap[wflow.getInputDataset()])
 
+
+    def getSecondaryDatasets(self, workflows):
```

Comment:
This method modifies workflows *in place* which is always dangerous business and not thread-safe. Instead takes the workflows and return a new list with proper parameter sets. In place changes can be so tricky to debug and not to mention that entire codebase should *knows* about it and have proper logic.

Reply:
I see, let me try to break it down into two pieces. There will be 3 other places requiring the same changes too.